### PR TITLE
[luci-interpreter] Support int8 FullyConnected

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This pr enables int8 FullyConnected operation.

Related issue: #7677

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@partner.samsung.com>

### CODE REVIEW

**Comment by binarman** [comment]
Linking problems, maybe try to rebase on master/older revision

**Comment by SlavikMIPT** [comment]
@nnfw-bot test nncc-release

File: compiler/luci-interpreter/src/kernels/FullyConnected.cpp [hunk 1]
```diff
@@ -77,6 +84,9 @@ void FullyConnected::execute() const
     case DataType::U8:
       evalQuantized();
```
Comment (m-bronnikov) [inline compiler/luci-interpreter/src/kernels/FullyConnected.cpp:85]: How about make `evalQuantized()` as template function for types with assymetric quantization like `int8` and `uint8`? =)

If so, we could reduce code duplication and use this method as:
```cpp
switch (input()->element_type())
  {
    case DataType::U8:
      evalQuantized<uint8_t>();
      break;
    case DataType::S8:
      evalQuantized<int8_t>();
      break;
    case DataType::FLOAT32:
      evalFloat();
      break;
```

**Review by m-bronnikov — APPROVED** [review]
LGTM

File: compiler/luci-interpreter/src/kernels/FullyConnected.cpp (hunk 1 above)
Comment (binarman) [reply compiler/luci-interpreter/src/kernels/FullyConnected.cpp:85]: This technique could be used for other operations, like Conv, DepthwiseConv, etc. Let's consider this refactoring after enabling minimal set (from #7677) of operations.

File: compiler/luci-interpreter/src/kernels/FullyConnected.test.cpp [hunk 2]
```diff
@@ -48,6 +48,39 @@ void Check(std::initializer_list<int32_t> input_shape, std::initializer_list<int
...
+template <>
+void Check<int8_t>(std::initializer_list<int32_t> input_shape,
+                   std::initializer_list<int32_t> weights_shape,
+                   std::initializer_list<int32_t> bias_shape,
+                   std::initializer_list<int32_t> output_shape,
+                   std::initializer_list<float> input_data,
+                   std::initializer_list<float> weights_data,
+                   std::initializer_list<float> bias_data, std::initializer_list<float> output_data)
+{
+  const float quantized_tolerance = getTolerance(-127, 128, 255);
```
Comment (binarman) [inline compiler/luci-interpreter/src/kernels/FullyConnected.test.cpp:60]: nitpick =)
minimum/maximum for int8 type is -128/127, not -127/128

**Review by binarman — APPROVED** [review]
LGTM

**Pull request merged**